prog/encoding.go

```diff
@@ -593,12 +597,18 @@ func (p *parser) parseArgString(t Type, dir Dir) (Arg, error) {
 		p.eatExcessive(true, "wrong string arg")
 		return t.DefaultArg(dir), nil
 	}
-	data, err := p.deserializeData()
+	data, b64, err := p.deserializeData()
```

Comment:
You need to decide and explicitly write to the documentation whether the fact of being encoded in base64 has to do with whether the data is actually compressed.

If these are unrelated and we need base64 just for a more compact representation of big binary data, you don't need to care about it here. Just fetch the raw bytes and then, if it's a compressed type, decompress them.

Reply:
Currently being encoded in Base64 means data is also compressed - but you are right that keeping them separate is cleaner, so I will update the implementation to match this.